disnake/ext/commands/params.py

```diff
@@ -1058,13 +1168,70 @@ def Param(
 param = Param
 
 
-def inject(function: Callable[..., Any]) -> Any:
+def inject(function: Callable[..., Any], *, autocompleters: Dict[str, Callable] = None) -> Any:
     """A special function to use the provided function for injections.
     This should be assigned to a parameter of a function representing your slash command.
 
     .. versionadded:: 2.3
+
+    .. versionchanged:: 2.6
+        Added ``autocompleters`` keyword-only argument.
+
+    Parameters
+    ----------
+    function: Callable
+        The injection function.
+    autocompleters: Dict[:class:`str`, Callable]
+        A mapping of the injection's option names to their respective autocompleters.
+
+        See also :func:`Injection.autocomplete`.
+
+        .. versionadded:: 2.6
+
+    Returns
+    -------
+    :class:`Injection`
+        The resulting injection
+
+        .. note::
+
+            In terms of typing, this returns ``Any`` to avoid typing issues caused by how this
+            extension work, but at runtime this is always an :class:`Injection` instance.
+            You can find more in-depth explanation :ref:`here <why_params_and_injections_return_any>`.
+    """
+
+    return Injection(function, autocompleters=autocompleters)
+
+
+def injection(*, autocompleters: Dict[str, Callable] = None) -> Callable[[Callable[..., Any]], Any]:
+    """Decorator interface for :func:`inject`.
+    You can then assign this value to your slash commands' parameters.
+
+    .. versionadded:: 2.6
+
+    Parameters
+    ----------
+    autocompleters: Dict[:class:`str`, Callable]
+        A mapping of the injection's option names to their respective autocompleters.
+
+        See also :func:`Injection.autocomplete`.
+
+    Returns
+    -------
+    Callable[[Callable[..., Any]], :class:`Injection`]
+        Decorator which turns your injection function into actual :class:`Injection`.
+
+        .. note::
+
+            In terms of typing, this returns ``Any`` to avoid typing issues caused by how this
+            extension work, but at runtime this is always an :class:`Injection` instance.
+            You can find more in-depth explanation :ref:`here <why_params_and_injections_return_any>`.
```

Comment:
```suggestion
            The decorator return type is annotated with ``Any`` to avoid typing issues caused by how this
            extension works, but at runtime this is always an :class:`Injection` instance.
            You can find more in-depth explanation :ref:`here <why_params_and_injections_return_any>`.
```